controllers/custom/custom_controller.go

```diff
@@ -170,7 +170,7 @@ func (c *CustomController) Start(ctx context.Context) error {
 func (c *CustomController) WaitForCacheSync(controller cache.Controller) {
 	for !controller.HasSynced() && controller.LastSyncResourceVersion() == "" {
 		c.log.Info("waiting for controller to sync")
-		time.Sleep(time.Second * 5)
```

Comment:
any drawbacks if we decrease the sleep time here?

Reply:
We will poll more. making this loop cpu intensive. On the other hand if we finish the sync right after we go to sleep, then we delay node reconciliation by 5 seconds. So 1 second seemed healthy tradeoff.